Documentation/EmacsConfiguration.md

```diff
@@ -0,0 +1,61 @@
+# Emacs Project Configuration
+
+Emacs can be configured with `lsp-mode` and `clangd` to work well.
+
+### clangd
+
+The official clangd extension can be used for C++ comprehension. You
+can use the following `.clangd` file placed in the project root:
+
+```yaml
+CompileFlags:
+  CompilationDatabase: Build/i686
+  Add:
+    - "-D__serenity__"
+    - "-UNO_TLS"
+    - "-I/path/to/serenity/Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0"
+    - "-I/path/to/serenity/Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0/i686-pc-serenity"
+```
+
+You will obviously need to update `11.2.0` to whatever your version at
+the time is.
+
+Run cmake at least once for this to work. clangd has difficulty
+finding specific methods and types, especially with inheritance
+trees. Also, include errors are wrong in 90% of cases.
+
+### lsp-mode
+
+```lisp
+(use-package lsp-mode
+  :hook ((c++-mode) . lsp-deferred)
+  :commands lsp
+  :config
+  (setq lsp-clients-clangd-args '("-j=4" "-background-index" "--log=error" "--clang-tidy" "--enable-config"))
+  (setq lsp-clangd-binary-path "/usr/bin/clangd"))
+
+```
+
+You can change the binary path to point to your local copy as shown,
+or your Serenity copy if you built the clang toolchain. Additionally,
```

Comment:
It's probably good to explain how to set the `clangd` binary path and where it is located. The "how" is `.dir-locals.el` and the "where" is `/serenityroot/Toolchain/Local/clang/bin/clangd`.

Reply:
Sure thing, I'll wait to here what you say about the previous comment regarding which clangd to use, then update accordingly.

Reply:
I didn't add this part since we don't have the instructions for building LLVM yet. Once we do I think it could be added.